tests/test-files.c

```diff
@@ -251,6 +251,32 @@ START_TEST(test_path_relative_child_01)
 END_TEST
 
 
+void
+test_set_absolute(const char *p, const char *expected)
+{
+    struct cork_path  *actual;
+
+    fprintf(stderr, "set_absolute(\"%s\") ?= \"%s\"\n",
+            (p == NULL)? "": p,
+            expected);
+
+    actual = cork_path_new(p);
+    cork_path_set_absolute(actual);
+    verify_path_content(actual, expected);
+    cork_path_free(actual);
+}
+
+START_TEST(test_path_set_absolute_01)
+{
+    /* We test that an absolute path really is so. */
+    DESCRIBE_TEST
+    test_set_absolute("/a", "/a");
+    test_set_absolute("/a/b", "/a/b");
+    test_set_absolute("/a/b/", "/a/b/");
```

Comment:
Use `cork_path_cwd` to get the current directory, and construct your expected strings from that via a `cork_buffer`.  (A helper macro might help alleviate the drudgery.  Don't worry about sharing the temp path or buffer instances, it's fine to create and destroy them willy-nilly, this is just a test suite.)